vm_to_cluster.rb

```diff
@@ -67,6 +67,12 @@ def data_check(cluster_lines)
   entries = parse_cluster_txt(cluster_txt)
 
   cluster_lines = entries.map do |e|
+    # HACK: We have not edited cluster.txt yet, but may have 
+    # forced the VMs to have a ${BACH_CLUSTER_PREFIX}-
+    # when we generated VM_LIST in vbox_create.sh
+    if ENV['BACH_CLUSTER_PREFIX'] != '' then
+      e[:hostname] = "#{ENV['BACH_CLUSTER_PREFIX']}-#{e[:hostname]}"
+    end
```

Comment:
As `vm-to-cluster.sh` has been able to be run outside of a cluster build, could this instead query Vagrant or VirtualBox to find what our cluster name is since we now have at least one VM (the bootstrap) which we know should always exist at this point?

I'm thinking we can get the cluster name via something like one of the following; also this could be dropped into `BACH::ClusterData::HypervisorNode`:
```ubuntu@dob2-bach-r4an07:~/third-cluster/chef-bach$ for v in $(vboxmanage list vms | sed -e 's/^.*{//' -e 's/}$//') ;do hits=$(vboxmanage showvminfo --machinereadable $v | grep $PWD) && echo -e "==== $(vboxmanage list vms | grep $v) ====\n$hits"; done
==== "thirdsies-bcpc-bootstrap" {dc7515e0-89d6-485a-8b7e-019a50ada510} ====
SharedFolderPathMachineMapping2="/home/ubuntu/third-cluster/chef-bach"
SharedFolderPathMachineMapping3="/home/ubuntu/third-cluster/chef-bach/vbox"
==== "thirdsies-bcpc-vm1" {a2c0e910-f970-40aa-a5b2-76e10553fb76} ====
CfgFile="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/thirdsies-bcpc-vm1.vbox"
SnapFldr="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots"
LogFldr="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Logs"
"SATA_Controller-0-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots/{d6965db9-9378-4720-90f5-28f9d56f3a55}.vdi"
"SATA_Controller-1-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots/{d21b9e84-9b02-45ce-a686-5354b38c75a8}.vdi"
"SATA_Controller-2-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots/{462f46b9-bf0c-49a6-b9d7-bc5e8fb7b9a6}.vdi"
"SATA_Controller-3-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots/{ee2144f0-31ae-4c2f-ab74-34cb5ef6b63b}.vdi"
"SATA_Controller-4-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots/{7bc7ef68-1f1f-440c-bc33-252d94776f08}.vdi"
"SATA_Controller-5-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/Snapshots/{f62b94e6-d69a-4823-ac53-773c4681a212}.vdi"
vcpfile="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm1/thirdsies-bcpc-vm1.webm"
==== "thirdsies-bcpc-vm2" {bdba5f4a-045f-4a51-972f-15b47eb542ea} ====
CfgFile="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/thirdsies-bcpc-vm2.vbox"
SnapFldr="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots"
LogFldr="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Logs"
"SATA_Controller-0-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots/{8223e2c1-b0ba-4cef-92a4-44f59d5277fe}.vdi"
"SATA_Controller-1-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots/{98e33779-2a62-4030-87e8-9d001bda44a0}.vdi"
"SATA_Controller-2-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots/{f32a8286-b003-4f57-98dc-13edc14ba8f5}.vdi"
"SATA_Controller-3-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots/{81b7a5fd-e247-43d4-8fe3-9626df162def}.vdi"
"SATA_Controller-4-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots/{55a80228-fd68-4df7-82d5-6ee47ed5062c}.vdi"
"SATA_Controller-5-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/Snapshots/{7dc13c42-a781-4c77-8976-4718095ef39d}.vdi"
vcpfile="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm2/thirdsies-bcpc-vm2.webm"
==== "thirdsies-bcpc-vm3" {7e3f59c6-d6d2-469c-93bd-6ebd81824b86} ====
CfgFile="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/thirdsies-bcpc-vm3.vbox"
SnapFldr="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots"
LogFldr="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Logs"
"SATA_Controller-0-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots/{bcf67827-4b68-49bb-a299-01b153578f1f}.vdi"
"SATA_Controller-1-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots/{8f4e9873-be27-4270-90e9-46bfafcbb44d}.vdi"
"SATA_Controller-2-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots/{09596bc8-59e0-43e2-be72-d6d312285829}.vdi"
"SATA_Controller-3-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots/{cfa6c275-2280-4cc5-a3b0-ef2246abacdd}.vdi"
"SATA_Controller-4-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots/{174b8679-500d-4adc-be38-8252af82e10b}.vdi"
"SATA_Controller-5-0"="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/Snapshots/{e34a782f-21de-4600-8655-a7f997424fab}.vdi"
vcpfile="/home/ubuntu/third-cluster/chef-bach/vbox/thirdsies-bcpc-vm3/thirdsies-bcpc-vm3.webm"
```

Or less code (but maybe not as stable of output):
```ubuntu@dob2-bach-r4an07:~/third-cluster/chef-bach$ vagrant status | grep bcpc-bootstrap | sed 's/[-]*bcpc-bootstrap.*//'
thirdsies```

Reply:
I don't understand why we need to do this when we have a simple environment variable or the lack of it that can inform our decision